src/OrchardCore.Modules/OrchardCore.Media/TagHelpers/ImageTagHelper.cs

```diff
@@ -0,0 +1,52 @@
+using System.Web;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using OrchardCore.Media.Processing;
+
+namespace OrchardCore.Media.TagHelpers
+{
+    [HtmlTargetElement("img", Attributes = ImageSrcAttributeName + "," + ImageSizeWidthAttributeName)]
+    [HtmlTargetElement("img", Attributes = ImageSrcAttributeName + "," + ImageSizeHeightAttributeName)]
+    [HtmlTargetElement("img", Attributes = ImageSrcAttributeName + "," + ImageSizeWidthAttributeName + "," + ImageSizeModeAttributeName)]
+    [HtmlTargetElement("img", Attributes = ImageSrcAttributeName + "," + ImageSizeHeightAttributeName + "," + ImageSizeModeAttributeName)]
+    public class ImageTagHelper : TagHelper
+    {
+        private const string ImageSizeAttributePrefix = "media-";
+
+        private const string ImageSrcAttributeName = ImageSizeAttributePrefix + "src";
+        private const string ImageSizeWidthAttributeName = ImageSizeAttributePrefix + "width";
+        private const string ImageSizeHeightAttributeName = ImageSizeAttributePrefix + "height";
+        private const string ImageSizeModeAttributeName = ImageSizeAttributePrefix + "resize-mode";
+
+        private readonly IMediaFileStore _mediaFileStore;
+
+        public ImageTagHelper(IMediaFileStore mediaFileStore)
+        {
+            _mediaFileStore = mediaFileStore;
+        }
+
+        [HtmlAttributeName(ImageSizeWidthAttributeName)]
+        public int? ImageWidth { get; set; }
+
+        [HtmlAttributeName(ImageSizeHeightAttributeName)]
+        public int? ImageHeight { get; set; }
+
+        [HtmlAttributeName(ImageSizeModeAttributeName)]
+        public ResizeMode ResizeMode { get; set; }
+
+        [HtmlAttributeName(ImageSrcAttributeName)]
+        public string Src { get; set; }
+
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            if (string.IsNullOrEmpty(Src))
+            {
+                return;
+            }
+
+            var resolvedSrc = _mediaFileStore != null ? _mediaFileStore.MapPathToPublicUrl(Src) : Src;
+            var resizedSrc = ImageSharpUrlFormatter.GetMediaResizeUrl(resolvedSrc, ImageWidth, ImageHeight, ResizeMode);
```

Comment:
If the `media-src` is not set then this shouldn't be called.

Reply:
I mean the `src` shouldn't be converted. Resizing and path conversion are independent.

Reply:
I'm not sure if I've understood correctly but...

If media-src isn't set then the tag helper wont be called.  media-src can be set by itself with no resize attributes and so is independent of resizing.

I thought about making media-src a separate helper which fired first and so making the asset directory resolution and resizing completely separate, however realised that image resizing only occurs for images in the media directory and therefor felt that resizing should be dependant on the use of media-src.

Does that make sense/sound reasonable?